src/util.ts

```diff
@@ -0,0 +1,66 @@
+import { execSync } from 'child_process'
+import * as os from 'os'
+
+export function validatePackageJson (packageJson : any) {
+  const s3PublishInfo = packageJson.build.publish; 
+  console.log('Detected publish info', JSON.stringify(s3PublishInfo, null, ' '))
+
+  let validationErrors: string[] = []
+  if (s3PublishInfo.provider !== "s3") {
+    throw new Error('package.json:publish.provider key must have value "s3"')
+  } else if (!s3PublishInfo.bucket) {
+    throw new Error('package.json:publish.bucket key must have a bucket name')
+  } else if (s3PublishInfo.path !== '${env.BRANCH}/${env.COMMIT}/${env.PLATFORM}') {
+    throw new Error('package.json:publish.path key must be set to "${env.BRANCH}/${env.COMMIT}/${env.PLATFORM}"')
+  }
+}
+
+export function getAppPlatform(): string {
+  if (process.env.APP_PLATFORM) {
```

Comment:
Why the conditional? Why not just return `process.env.APP_PLATFORM || process.env.PLATFORM || process.env.TRAVIS_OS_NAME || os.platform()`

Reply:
So that I can overwrite it for local testing. Also I was thinking it could be potentially useful for supporting other CI's if one knows what env vars are needed.